### PR TITLE
kv/client: decrease standby gRPC connection count

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

In an internal test, TiCDC replicates 1300 tables from upstream TiDB to downstream. We found the goroutine count is far more than we expect. After dump the goroutine profile, we found the gRPC related goroutines count reaches 54588.

The more connections established, the more CPU and memory will be wasted.

```
goroutine profile: total 110989
13668 @ 0x43b2a5 0x44c517 0xab834c 0x472ee1
#	0xab834b	google.golang.org/grpc.(*ccBalancerWrapper).watcher+0xab	google.golang.org/grpc@v1.27.1/balancer_conn_wrappers.go:69

13652 @ 0x43b2a5 0x40692f 0x40656b 0xac0385 0x472ee1
#	0xac0384	google.golang.org/grpc.(*addrConn).resetTransport+0x464	google.golang.org/grpc@v1.27.1/clientconn.go:1179

13652 @ 0x43b2a5 0x44c517 0xa300df 0xa30a3d 0xa5191b 0x472ee1
#	0xa300de	google.golang.org/grpc/internal/transport.(*controlBuffer).get+0xfe	google.golang.org/grpc@v1.27.1/internal/transport/controlbuf.go:395
#	0xa30a3c	google.golang.org/grpc/internal/transport.(*loopyWriter).run+0x1dc	google.golang.org/grpc@v1.27.1/internal/transport/controlbuf.go:513
#	0xa5191a	google.golang.org/grpc/internal/transport.newHTTP2Client.func3+0x7a	google.golang.org/grpc@v1.27.1/internal/transport/http2_client.go:346

13650 @ 0x43b2a5 0x44c517 0xa40734 0x472ee1
#	0xa40733	google.golang.org/grpc/internal/transport.(*http2Client).keepalive+0x173	google.golang.org/grpc@v1.27.1/internal/transport/http2_client.go:1331

13618 @ 0x43b2a5 0x4335bb 0x46c9f5 0x4e9145 0x4ea235 0x4ea217 0x54f42f 0x563771 0x51d6a2 0x4e0467 0xa217a9 0xa21742 0xa22025 0xa3fdc5 0x472ee1
#	0x46c9f4	internal/poll.runtime_pollWait+0x54					runtime/netpoll.go:222
#	0x4e9144	internal/poll.(*pollDesc).wait+0x44					internal/poll/fd_poll_runtime.go:87
#	0x4ea234	internal/poll.(*pollDesc).waitRead+0x1d4				internal/poll/fd_poll_runtime.go:92
#	0x4ea216	internal/poll.(*FD).Read+0x1b6						internal/poll/fd_unix.go:166
#	0x54f42e	net.(*netFD).Read+0x4e							net/fd_posix.go:55
#	0x563770	net.(*conn).Read+0x90							net/net.go:183
#	0x51d6a1	bufio.(*Reader).Read+0x221						bufio/bufio.go:227
#	0x4e0466	io.ReadAtLeast+0x86							io/io.go:328
#	0xa217a8	io.ReadFull+0x88							io/io.go:347
#	0xa21741	golang.org/x/net/http2.readFrameHeader+0x21				golang.org/x/net@v0.0.0-20210316092652-d523dce5a7f4/http2/frame.go:237
#	0xa22024	golang.org/x/net/http2.(*Framer).ReadFrame+0xa4				golang.org/x/net@v0.0.0-20210316092652-d523dce5a7f4/http2/frame.go:492
#	0xa3fdc4	google.golang.org/grpc/internal/transport.(*http2Client).reader+0x184	google.golang.org/grpc@v1.27.1/internal/transport/http2_client.go:1265
```

more goroutines:
[goroutine.txt](https://github.com/pingcap/ticdc/files/6574667/goroutine.txt)

### What is changed and how it works?

The goroutine used in gRPC connection is as expected, because for each TiKV store, TiCDC creates a connection array, which contains `10 connections` by default.

After investigation, the connection pool can be decrease, for two reasons:

- gRPC itself has builtin error handling and backoff mechanism.
- TiCDC only uses one connection from the connection pool at the same time, we don't need too many backup connections.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Decrease the default gRPC connection pool size to decrease goroutines count.

### CODE REVIEW

**Review by leoppro — APPROVED** [review]

**Review by ben1009 — APPROVED** [review]

File: cdc/kv/client.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ const (
 	grpcInitialWindowSize     = 1 << 28 // 256 MB The value for initial window size on a stream
 	grpcInitialConnWindowSize = 1 << 29 // 512 MB The value for initial window size on a connection
 	grpcMaxCallRecvMsgSize    = 1 << 30 // 1024 MB The maximum message size the client can receive
-	grpcConnCount             = 10
+	grpcConnCount             = 2
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/kv/client.go:64]: Seems we can't control the parameter in server side

**Review by leoppro — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**